From: Or Gerlitz <ogerlitz@mellanox.com>
Date: Thu, 14 Jul 2016 10:32:43 +0300
Subject: net/switchdev: Export the same parent ID service function
Patch-mainline: v4.8-rc1
Git-commit: 8438884d4ab423161b974854ebb90c08219dd678
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

This helper serves to know if two switchdev port netdevices belong to the
same HW ASIC, e.g to figure out if forwarding offload is possible between them.

Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 include/net/switchdev.h   |    8 ++++++++
 net/switchdev/switchdev.c |    5 +++--
 2 files changed, 11 insertions(+), 2 deletions(-)

--- a/include/net/switchdev.h
+++ b/include/net/switchdev.h
@@ -208,6 +208,8 @@ void switchdev_port_fwd_mark_set(struct
 				 struct net_device *group_dev,
 				 bool joining);
 
+bool switchdev_port_same_parent_id(struct net_device *a,
+				   struct net_device *b);
 #else
 
 static inline void switchdev_deferred_process(void)
@@ -332,6 +334,12 @@ static inline void switchdev_port_fwd_ma
 {
 }
 
+static inline bool switchdev_port_same_parent_id(struct net_device *a,
+						 struct net_device *b)
+{
+	return false;
+}
+
 #endif
 
 #endif /* _LINUX_SWITCHDEV_H_ */
--- a/net/switchdev/switchdev.c
+++ b/net/switchdev/switchdev.c
@@ -1268,8 +1268,8 @@ void switchdev_fib_ipv4_abort(struct fib
 }
 EXPORT_SYMBOL_GPL(switchdev_fib_ipv4_abort);
 
-static bool switchdev_port_same_parent_id(struct net_device *a,
-					  struct net_device *b)
+bool switchdev_port_same_parent_id(struct net_device *a,
+				   struct net_device *b)
 {
 	struct switchdev_attr a_attr = {
 		.id = SWITCHDEV_ATTR_ID_PORT_PARENT_ID,
@@ -1303,6 +1303,7 @@ static u32 switchdev_port_fwd_mark_get(s
 
 	return dev->ifindex;
 }
+EXPORT_SYMBOL_GPL(switchdev_port_same_parent_id);
 
 static void switchdev_port_fwd_mark_reset(struct net_device *group_dev,
 					  u32 old_mark, u32 *reset_mark)
